### PR TITLE
[9.4] ISPN-10127 Unquote quoted site names in the CLI

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-10127

### CODE REVIEW

File: cli/cli-interpreter/src/test/java/org/infinispan/cli/interpreter/SiteStatementTest.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public void testSiteStatus() throws Exception {
       String nycCache = cache("NYC", 0).getName();
       String nycSessionId = nycInterpreter.createSessionId(nycCache);
 
-      assertInterpreterOutput(lonInterpreter, lonSessionId, "site --status NYC;", "online");
+      assertInterpreterOutput(lonInterpreter, lonSessionId, "site --status \"NYC\";", "online");
```
Comment (ryanemerson) [inline cli/cli-interpreter/src/test/java/org/infinispan/cli/interpreter/SiteStatementTest.java]: We should modify this test, or add a new one, that tests against a site name with an actual hyphen in it.

**Comment by galderz** [comment]
@tristantarrant I agree with @ryanemerson here, can you provide a test for this?

**Comment by tristantarrant** [comment]
I have gone a bit further than that. XSite tests now use proper parameterized site names, and they now all default to having hypens in them :) I will need to forward port these changes to master if you're ok with them.

**Comment by tristantarrant** [comment]
Fixed AsyncOfflineTest

**Pull request merged**